helper-cli/src/main/kotlin/commands/CreateAnalyzerResultFromPackageListCommand.kt

```diff
@@ -0,0 +1,168 @@
+
+import com.fasterxml.jackson.module.kotlin.readValue
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.file
+
+import org.ossreviewtoolkit.helper.utils.writeOrtResult
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Repository
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.Excludes
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.config.ScopeExclude
+import org.ossreviewtoolkit.model.config.ScopeExcludeReason
+import org.ossreviewtoolkit.model.mapper
+import org.ossreviewtoolkit.model.orEmpty
+import org.ossreviewtoolkit.utils.common.expandTilde
+import org.ossreviewtoolkit.utils.ort.Environment
+
+internal class CreateAnalyzerResultFromPackageListCommand : CliktCommand(
+    "A command which turns a simple definition file into an analyzer result."
+) {
+    private val packageListFile by option(
+        "--package-list-file", "-i",
+        help = "The package list file to read the packages metadata and the project metadata from."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val ortFile by option(
+        "--ort-file", "-o",
+        help = "The ORT file to write the generated analyzer result to."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        val packageList = packageListFile.mapper().readValue<PackageList>(packageListFile)
+
+        val projectVcs = VcsInfo(
+            type = packageList.vcsType ?: VcsType.UNKNOWN,
+            url = packageList.vcsUrl.orEmpty(),
+            revision = packageList.vcsRevision.orEmpty(),
+            path = packageList.vcsPath.orEmpty()
+        )
+
+        val projectName = packageList.name?.takeUnless { it.isBlank() } ?: DEFAULT_PROJECT_NAME
+
+        val project = Project.EMPTY.copy(
+            id = Identifier("Unmanaged::$projectName:"),
```

Comment:
This still hard-codes "Unmanged".

Reply:
I kept it hard-coded, because the running the code snippet I copied from your comment leads to an unsuccessful look-up in the map. I haven't investigated the reason...

Reply:
Would a code comment work for you as well?

Reply:
It's a compromise that I can live with.